pandas/io/common.py

```diff
@@ -1207,25 +1212,22 @@ def _maybe_memory_map(
 
     # need to open the file first
     if isinstance(handle, str):
-        if encoding and "b" not in mode:
-            # Encoding
-            handle = open(handle, mode, encoding=encoding, errors=errors, newline="")
-        else:
-            # Binary mode
-            handle = open(handle, mode)
+        handle = open(handle, "rb")
```

Comment:
mmap is used only for reading and there should be no point in opening the file in text mode as mmap should bypass that.